torch/nn/parallel/distributed.py

```diff
@@ -253,38 +291,75 @@ def _queue_reduction(self, bucket_idx):
             self.reduced = [False] * len(self.bucket_sizes)
 
             def sync_reduction_streams():
-                # We only have to sync with the first one, but it's safer to do it this way
-                # in case we change the way in which we paralellize work
-                r_streams = zip(*self._reduction_streams)
-                for dev_id, default_stream, dev_r_streams in zip(self.device_ids, self._default_streams, r_streams):
-                    with torch.cuda.device(dev_id):
-                        for reduction_stream in dev_r_streams:
-                            default_stream.wait_stream(reduction_stream)
+                # We sync on the default streams for NCCL backend since all
+                # nccl reduction kernels goes to the default streams
```

Comment:
Can't you just make `reduction_streams` point to default streams for NCCL? This would allow you to use the same code